### PR TITLE
(Cleanup) Turn IBasePickerSuggestionsProps into a Pick<> type

### PR DESCRIPTION
**Pull request by Adjective-Object**
#### Pull request checklist

- [ ] ~~Addresses an existing issue: Fixes #0000~~
- [x] Include a change request file using `$ npm run change`

#### Description of changes

IBasePickerSuggestionProps is a manually declared slice of ISuggestionProps, so I turned it into an actual slice to avoid the types diverging over time & to more clearly express that it is a slice of another type.

It looks like they actually have diverged already:
https://github.com/OfficeDev/office-ui-fabric-react/blob/236b2eedeedc036b036327abf2723fbc3a3b3155/packages/office-ui-fabric-react/src/components/pickers/Suggestions/Suggestions.types.ts#L172-L180

LMK how you'd like to handle this conflict. I think that `() => JSX.Element` is a subtype of `(props: ISuggestionsProps<T>) => JSX.Element` so this is a non-breaking api contract change.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8054)

### CODE REVIEW

File: packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.ts [hunk 1]
```diff
@@ -1925,24 +1925,7 @@ interface IBasePickerStyles {
...
-  resultsMaximumNumber?: number;
-  searchForMoreText?: string;
-  searchingText?: string;
-  showRemoveButtons?: boolean;
-  suggestionsAvailableAlertText?: string;
-  suggestionsClassName?: string;
-  suggestionsContainerAriaLabel?: string;
-  suggestionsHeaderText?: string;
-  suggestionsItemClassName?: string;
+interface IBasePickerSuggestionsProps<T = any> extends Pick<ISuggestionsProps<TextDecoderOptions>, 'onRenderNoResultFound' | 'suggestionsHeaderText' | 'mostRecentlyUsedHeaderText' | 'noResultsFoundText' | 'className' | 'suggestionsClassName' | 'suggestionsItemClassName' | 'searchForMoreText' | 'forceResolveText' | 'loadingText' | 'searchingText' | 'resultsFooterFull' | 'resultsFooter' | 'resultsMaximumNumber' | 'showRemoveButtons' | 'suggestionsAvailableAlertText' | 'suggestionsContainerAriaLabel'> {
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.ts]: I'm curious if you've been able to verify this truly appears as a 100% compatible API change to consumers?

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts [hunk 2]
```diff
@@ -188,92 +188,27 @@ export interface IBasePickerProps<T> extends React.Props<any> {
...
-   * Screen reader message to read when there are suggestions available.
-   */
-  suggestionsAvailableAlertText?: string;
-
-  /**
-   * An ARIA label for the container that is the parent of the suggestions.
-   */
-  suggestionsContainerAriaLabel?: string;
-}
+export interface IBasePickerSuggestionsProps<T = any>
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts]: I don't think `T` is actually used for any of the picked props, is it? If not, I think it should just be removed here.

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts [hunk 3]
```diff
@@ -188,92 +188,27 @@ export interface IBasePickerProps<T> extends React.Props<any> {
...
+  extends Pick<
+    ISuggestionsProps<TextDecoderOptions>,
```
[the remark this answers is not in the document]
Comment (Adjective-Object) [reply packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts]: yeah, this is definitely an unfortunate autocomplete.

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts (hunk 2 above)
Comment (Adjective-Object) [reply packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts]: oh actually it is required.

Like i mentioned in the description, `'resultsFooterFull'` and `'resultsFooter'` are narrowed to not take the suggestion prop of the item into account when rendering. (hence diverging from the picked type).

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Adjective-Object) [reply packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts]: Yep

File: packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.ts (hunk 1 above)
Comment (Adjective-Object) [reply packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.ts]: Manually & by checking against API consumers in examples.

But like I said in the description, it diverges for resultsFooterFull and resultsFooter.

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts (hunk 2 above)
Comment (Adjective-Object) [reply packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts]: If we were to redefine the type as 
```typescript
/**
 * Subset of picker options that may be legally passed through a picker to its
 * internal Suggestions component.
 */
export interface IBasePickerSuggestionsProps<T = any>
  extends Pick<
    ISuggestionsProps<T>,
    | 'onRenderNoResultFound'
    | 'suggestionsHeaderText'
    | 'mostRecentlyUsedHeaderText'
    | 'noResultsFoundText'
    | 'className'
    | 'suggestionsClassName'
    | 'suggestionsItemClassName'
    | 'searchForMoreText'
    | 'forceResolveText'
    | 'loadingText'
    | 'searchingText'
    | 'resultsMaximumNumber'
    | 'showRemoveButtons'
    | 'suggestionsAvailableAlertText'
    | 'suggestionsContainerAriaLabel'
  > {
  /**
   * A renderer that adds an element at the end of the suggestions list it has more items than resultsMaximumNumber.
   */
  resultsFooterFull?: () => JSX.Element;

  /**
   * A renderer that adds an element at the end of the suggestions list it has fewer items than resultsMaximumNumber.
   */
  resultsFooter?: () => JSX.Element;
}
```

Then there would be no problem

Comment (Adjective-Object) [reply packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts]: (But this isn't 100% reflective of the actual requirements of the component)

File: packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.ts [hunk 4]
```diff
@@ -11575,7 +11558,7 @@ class NormalPeoplePickerBase extends BasePeoplePicker {
   static defaultProps: {
     createGenericItem: typeof createGenericItem;
     onRenderItem: (props: IPeoplePickerItemSelectedProps) => JSX.Element;
-    onRenderSuggestionsItem: (personaProps: IPersonaProps, suggestionsProps?: IBasePickerSuggestionsProps | undefined) => JSX.Element;
+    onRenderSuggestionsItem: (personaProps: IPersonaProps, suggestionsProps?: IBasePickerSuggestionsProps<any> | undefined) => JSX.Element;
```
[the remark this answers is not in the document]
Comment (Adjective-Object) [reply packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.ts]: On second look, this is in the generated type contract, and is just the result of compiling the declared types. It should be compatible with the existing declaration.

[the remark this answers is not in the document]
Comment (JasonGore) [reply packages/office-ui-fabric-react/etc/office-ui-fabric-react.api.ts]: Sorry, I missed that too.

**Comment by dzearing** [comment]
@JasonGore any other comments?

**Pull request merged**